src/lib/rf_data_recording_api_def.py

```diff
@@ -377,10 +379,45 @@ def round_up_to_even(f):
                 args_out = args_in + ",master_clock_rate=184.32e6"
             else:
                 args_out = args_in + ",master_clock_rate=200e6"
+        
+        # Derive master clock rate for X410 USRP
+        if "X4" in usrp_mboard_id:

```

Comment:
Please make it as if....elif ..else
In the current implementation: If X3 ... if X4 ...else .... If the USRP is X310, its arg will be overwritten by the arg under else of X410